src/manager.cpp

```diff
@@ -93,6 +93,121 @@ uint32_t SessionManager::closeAllByType(SessionType type)
     return count;
 }
 
+uint32_t SessionManager::closeUserSessionsByType(SessionType type,
+                                                 std::string sessionId,
+                                                 bool ownedOnly)
+{
+    auto numSessId = parseSessionId(sessionId);
+    std::string userName = getSessionItem(sessionId)->getOwner();
+    bool allowed = isAllSessionsAllowed(sessionId);
+    const auto count =
+        std::erase_if(sessionItems, [allowed, ownedOnly, type, userName,
+                                     numSessId](const auto& item) {
+            auto const [id, session] = item;
```

Comment:
Use `const auto& [id, session] = item;` to avoid unnecessary copying  of objects.

Reply:
@nest1ing Fixed